### PR TITLE
Added Busy Indicator

### PR DESCRIPTION
**Pull request by angelcalvasp**
Still pending Foreground and Background color
added sample view

### CODE REVIEW

**Comment by BSick7** [comment]
It would be best to isolate responsibilities.  The `BusyIndicator` typically has the following responsibilities:
- display progress (indeterminate or 0 to 100%)
- ability to mask section of UI

The custom XAML to make a spinner could be abstracted into a custom style for `ProgressBar` indeterminate.

**Comment by angelcalvasp** [comment]
Ok, the appropriate changes will be made

**Comment by angelcalvasp** [comment]
Changes have been made need help with abstraction of a custom style

**Comment by angelcalvasp** [comment]
OverlayStyle has been added, still pending Styling of content 

- inherits from ContentControl : done
- allows restyling of ProgressBar instead of taking on ProgressBar properties (value, min, max) : done
-defaults to spinner : done
-allows styling of overlay : done
-allows styling of content pending

**Comment by angelcalvasp** [comment]
Will do

File: src/BusyIndicator.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+module Fayde.Controls {
+    export class BusyIndicator extends ContentControl {
+        static IsBusyProperty = DependencyProperty.Register("IsBusy", () => Boolean, BusyIndicator, false);
+        static OverlayBrushProperty = DependencyProperty.Register("OverlayBrush", () => Media.Brush, BusyIndicator);
+        static HasContentProperty = DependencyProperty.RegisterReadOnly("HasContent", () => Boolean, BusyIndicator, false);
+        static BusyContentProperty = DependencyProperty.Register("BusyContent", () => Object, BusyIndicator, undefined, (d: BusyIndicator, args) => d.OnBusyContentChanged(args.OldValue, args.NewValue));
+        static BusyContentTemplateProperty = DependencyProperty.Register("BusyContentTemplate", () => DataTemplate, BusyIndicator);
+        static OverlayStyleProperty = DependencyProperty.Register("OverlayStyle",() => Style,BusyIndicator);
+        IsBusy: boolean;
+        OverlayBrush: Media.Brush;
```
Comment (BSick7) [inline src/BusyIndicator.ts]: This should be `OverlayStyle: Style`.

File: testsite/Views/busyindicator.fayde [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            <ColumnDefinition Width="400"/>
+            <ColumnDefinition Width="400"/>
+            <ColumnDefinition Width="400"/>
+        </Grid.ColumnDefinitions>
+        <Grid.RowDefinitions>
+            <RowDefinition Height="50"/>
+            <RowDefinition Height="*"/>
+        </Grid.RowDefinitions>
+        <TextBlock Text="Default settings" Grid.Column="0" Grid.Row="0" HorizontalAlignment="Center" VerticalAlignment="Center"/> 
+        <controls:BusyIndicator Grid.Column="0" Grid.Row="1" IsBusy="True"/>
```
Comment (BSick7) [inline testsite/Views/busyindicator.fayde]: Let's make these tests have toggles to turn on/off `IsBusy`.

**Comment by angelcalvasp** [comment]
Will do

**Comment by angelcalvasp** [comment]
Changes have been made

**Pull request merged**